index.js

```diff
@@ -23,13 +23,16 @@ ${footer}`
 
   const processAsset = async (bundle, processFn) => {
     const {name} = bundle
-    const wrappingCode = await processFn({name, bundler})
 
-    if (wrappingCode) {
-      writeAsset(name, wrappingCode)
+    if (name !== undefined) {
```

Comment:
Thank you very much, super fast :)
Does this mean that if `bundle.name` is not set the plugin won't do anything?
Is `bundle.name` always undefined when using multiple entry points?
Not following much the development of Parcel right now.
I'm happy to merge this one but could you clarify the above points?
Thank you again

Reply:
Hey Albino,
Sure thing. Having an `undefined` bundle name leads to a catch there.
https://github.com/albinotonnina/parcel-plugin-wrapper/blob/95d7a81bed0b2f36b14b72cfea9baf95e29e6ea8/index.js#L44-L48

As far as I have tested `bundle.name` is always `undefined` when there are multiple entry points. Whereas, `bundle.childBundles` contains these multiple entry points with correct `name` values.

Reply:
I see, thank you!
I guess traversing the childBundles and applying the plugin to each of the output files could be something to explore?
Feels like a different PR tho. Is it something worth considering anyway?

Reply:
> Is `bundle.name` always undefined when using multiple entry points?

Parcel creates an unnamed container bundle that contains the multiple entry points. It looks like the `childBundles` are the actual entry points.

Reply:
> I see, thank you!
> I guess traversing the childBundles and applying the plugin to each of the output files could be something to explore?
> Feels like a different PR tho. Is it something worth considering anyway?

This recursion is already doing it when there are multiple entry points
https://github.com/albinotonnina/parcel-plugin-wrapper/blob/95d7a81bed0b2f36b14b72cfea9baf95e29e6ea8/index.js#L32-L34

And note that calling this line regardless is crucial. For example entry point might be an `index.html` with the following content:

```
<div id="root">
</div>
<script type="text/javascript" src="./index.js"></script>
```

In this case parcel creates an output similar to this:

```
<div id="root"> 
</div><script type="text/javascript" src="/ui.cdfc7045.js"></script>
```

Not traversing `bundle.childBundles` would result in `ui.cdfc7045.js` being omitted.
This is why `forEach` function called on `bundle.childBundles` is not in `else` of the `if` statement, but outside.